### PR TITLE
Replace FilterNode::Expression with RowExpression

### PR DESCRIPTION
**Pull request by highker**
copied from #12523. Github accidentally closed my PR and disallow me to reopen it.

### CODE REVIEW

**Comment by highker** [comment]
@rschlussel any extra comments on this?

**Comment by highker** [comment]
Thanks everyone for the review! All comments addressed. I will land this once the test lights green.

File: presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java [hunk 1]
```diff
@@ -1201,6 +1203,22 @@ private PhysicalOperation visitScanFilterAndProject(
                 sourceLayout = source.getLayout();
             }
 
+            // filterExpression may contain large function calls; evaluate them before compiling.
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java]: Curious: Is this part of optimization newly enabled by `RowExpression` ? :)

Comment (highker) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java]: Is was there in sqlToRowExpression translation opt mode.

Comment (highker) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java]: I will need to extract this to a common function. Ideally whenever `toRowExpression` was called originally, this common function should be called to optimized the expression.

Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java]: Update:  `LocalExecutionPlanner.toRowExpression` calls `SqlToRowExpressionTranslator.translate` with `optimize=true`: 

https://github.com/prestodb/presto/blob/2e199b0e64b346c6641b67b70ee442f1b0617c76/presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java#L1289-L1292

Which tries to use `ExpressionOptimizer` to optimize the generated `RowExpression`:

https://github.com/prestodb/presto/blob/2e199b0e64b346c6641b67b70ee442f1b0617c76/presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java#L168-L171

Looks like `ExpressionOptimizer` has some basic `RowExpression` evaluation ability to support basic optimizations such as constant folding (over `RowExpression`): https://github.com/prestodb/presto/blob/2e199b0e64b346c6641b67b70ee442f1b0617c76/presto-main/src/main/java/com/facebook/presto/sql/relational/optimizer/ExpressionOptimizer.java 

This is now done by the new introduced `RowExpressionInterpreter` :)

Comment (highker) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java]: Fixed in #12702

**Pull request merged**